src/Options/Generic.hs

```diff
@@ -395,15 +398,43 @@ class ParseField a where
         -> Parser [a]
     parseListOfField h m c = many (parseField h m c)
 
+    readField :: ReadM a
+    default readField :: Read a => ReadM a
+    readField = auto
+
 instance ParseField Bool
 instance ParseField Double
 instance ParseField Float
-instance ParseField Int
 instance ParseField Integer
 instance ParseField Ordering
 instance ParseField ()
 instance ParseField Void
 
+readIntegralBounded :: forall a. (Integral a, Bounded a) => ReadM a
+readIntegralBounded =
+    auto >>= f
+    where
+        f i | i < toInteger (minBound :: a) = empty
```

Comment:
Could you add more descriptive error messages for these field parse failures?

(Same comment for other `empty`s)